scality_manila_utils/cli.py

```diff
@@ -0,0 +1,190 @@
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument(
+        '--exports',
+        help='Path to exports file',
+        default='/etc/exports.conf'
+    )
+    pre_parser.add_argument(
+        '--root-export',
+        help='NFS export path to the ring NFS root volume',
+        default='127.0.0.1:/'
+    )
+    pre_parser.add_argument(
+        '--debug',
+        help='Set debug log level',
+        action='store_true',
+        default=False
+    )
+
+    command_parser = argparse.ArgumentParser(
+        parents=[pre_parser],
+        description='Manila exports management'
+    )
+    subparsers = command_parser.add_subparsers(help='sub-command help')
+
+    parser_create = subparsers.add_parser(
+        'create',
+        help='Prepare an export without any access grants'
+    )
+    parser_create.add_argument(
+        'export_name',
+        help='Export point to create'
+    )
+
+    parser_grant = subparsers.add_parser(
+        'grant',
+        help='Grant access to an existing filesystem, and reexport it'
+    )
+    parser_grant.add_argument(
+        'export_name',
+        help='Filesystem to grant access to'
+    )
+    parser_grant.add_argument(
+        'host',
+        help='IP address or network to grant access for'
+    )
+    parser_grant.add_argument(
+        'options',
+        help='Export options',
+        nargs='*'
+    )
+
+    parser_revoke = subparsers.add_parser(
+        'revoke',
+        help='Revoke access from an existing filesystem, and reexport it'
+    )
+    parser_revoke.add_argument(
+        'export_name',
+        help='Filesystem to revoke access from'
+    )
+    parser_revoke.add_argument(
+        'host',
+        help='IP address or network to revoke access for'
+    )
+
+    parser_check = subparsers.add_parser(
+        'check',
+        help='Check for required binaries and running services'
+    )
+
+    parser_create.set_defaults(func=Helper.add_export)
+    parser_grant.set_defaults(func=Helper.grant_access)
+    parser_revoke.set_defaults(func=Helper.revoke_access)
+    parser_check.set_defaults(func=Helper.verify_environment)
+
+    parsed_args = command_parser.parse_args(args)
+
+    # Set debug level if requested
+    if parsed_args.debug:
+        logging.root.setLevel(logging.DEBUG)
+
+    # Drop any elevated permissions
+    drop_privileges()
+
+    # Setup exports helper
+    try:
+        helper = Helper(parsed_args.root_export, parsed_args.exports)
+    except EnvironmentException:
+        log.exception("Unable to setup")
+        raise
+
+    command_args = dict(
+        (k, v) for k, v in vars(parsed_args).items()
+        if k not in ('exports', 'root_export', 'func', 'debug')
+    )
+
+    formatted_args = ", ".join(
+        "{arg:s}={val!r}".format(arg=arg, val=val)
+        for arg, val in command_args.items()
+    )
+    log.info("Invoking %s(%s)", parsed_args.func.__name__, formatted_args)
+    try:
```

Comment:
Instead of having a couple of `try`/`except` clauses inside `main` which then `log.exception` and `raise`, I'd wrap `main` in a decorator which does this for any (unhandled) exception.